tests/chainer_tests/links_tests/normalization_tests/test_batch_normalization.py

```diff
@@ -37,118 +29,206 @@ def _batch_normalization(expander, gamma, beta, x, mean, var, eps, test):
         {'input_shape': (5, 4, 3), 'axis': (0, 1)},
     ]
 )))
-class BatchNormalizationTest(unittest.TestCase):
+
+
+_inject_backend_tests = testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+        {'use_ideep': 'always'},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+
+
+def _batch_normalization(expander, gamma, beta, x, mean, var, eps, test):
+    mean = mean[expander]
+    if test:
+        std = numpy.sqrt(var[expander])
+    else:
+        std = numpy.sqrt(var[expander] + eps)
+    y_expect = gamma * (x - mean) / std + beta
+    return y_expect
+
+
+class BatchNormalizationTestBase(testing.LinkTestImpl):
```

Comment:
Implementing in inheritance, I don't think `LinkTestImpl` is needed: the base implementation can simply inherit from `object`.

Reply:
Fixed.